### DISCUSSION TITLE
How to install Reflexil Plugin on macOS?

### DISCUSSION THREAD

**Discussion by StupidRepo** [icsharpcode/ILSpy#2430] [Help]
Now, your wondering:

> How is this guy using ILSpy on macOS?

Using [this ILSpy build](https://github.com/icsharpcode/AvaloniaILSpy/releases/tag/v7.0-rc1), I can use it on macOS.

So anyways, I put the Reflexil folder contents in the ILSpy/Contents/MacOS folder but I don't think it worked?

Here is what I am trying to do anyways:

I need a DNSpy that works for MacOS so that I can edit game code so I chose ILSpy but then I had to choose a [macOS-Compatible ILSpy build](https://github.com/icsharpcode/AvaloniaILSpy/releases/tag/v7.0-rc1).

Turns out, ILSpy can't decompile, edit code and then recompile. So anyways, I found the Reflexil plugin which (correct me if I am wrong) allows you to decompile, edit code and recompile.

I dragged all the folder contents into the ILSpy/Contents/MacOS folder, relaunched ILSpy but I don't think anything changed.

Please help me to achieve this.

(Note: I can't use Wine with DNSpy, it breaks DNSpy's UI so much.)

**Comment by christophwille** [comment]
AvaloniaILSpy is in no way, shape or form tested for compatibility of Windows ILSpy (they are on different UI stacks to start with)

**Comment by StupidRepo** [comment]
Ah ok.

**Comment by StupidRepo** [comment]
How should I install it though?

**Comment by christophwille** [comment]
You knew this was coming - use ILSpy on Windows.

**Comment by siegfriedpammer** [comment]
> AvaloniaILSpy is in no way, shape or form tested for compatibility of Windows ILSpy (they are on different UI stacks to start with)

To make Christoph's words more explicit: AvaloniaILSpy is a "fork"/port of ILSpy based on Avalonia, however, they only share the decompiler engine. All other parts are completely different and NOT binary-compatible, thus it is impossible to get any plugin for ILSpy to work with AvaloniaILSpy.

**Comment by StupidRepo** [comment]
> You knew this was coming - use ILSpy on Windows.

I use an M1 Mac (so bootcamp won't work) and I have no Windows PC...

**Comment by christophwille** [comment]
Parallels should do the trick.
